addons/folo/common/src/main/java/org/commonjava/indy/folo/change/FoloTrackingListener.java

```diff
@@ -118,6 +118,18 @@ public void onFileAccess( @Observes final FileAccessEvent event )
             logger.trace( "Tracking report: {} += {} in {} (DOWNLOAD)", trackingKey, trackingPath,
                           keyedLocation.getKey() );
 
+            /*
+             * If a build makes a request to indy-admin service, pnc sends it through generic proxy
+             * where it gets tracked once and then it gets tracked second time on the indy-admin service.
+             * We should avoid the tracker if it sends request through generic proxy and the target is
+             * indy instance,
+             */
+            if ( trackingPath.contains("api/folo/track") )
```

Comment:
Where does this "api/folo/track" come from? if it is from "ORIGIN_PATH", maybe we can pull it to the beginning of this method to avoid any following logic handling. Like:
if (metadata.get(ORIGIN_PATH)!=null && metadata.get(ORIGIN_PATH).contains("api/folo/track") )) {
logger.....
return 
}

Reply:
Yeah, that's a good point, this is from original request which can be handled as early as possible.